app/org/sagebionetworks/bridge/json/ScheduleDeserializer.java

```diff
@@ -33,19 +36,23 @@ public Schedule deserialize(JsonParser parser, DeserializationContext context) t
         Schedule schedule = new Schedule();
         if (node.has(ACTIVITIES_PROPERTY)) {
             schedule.setActivities(JsonUtils.asEntityList(node.get(ACTIVITIES_PROPERTY), Activity.class));
-        } else if (node.has(ACTIVITY_TYPE_PROPERTY) && node.has(ACTIVITY_REF_PROPERTY)) {
-            String label = JsonUtils.asText(node, LABEL_PROPERTY);
-            String ref = JsonUtils.asText(node, ACTIVITY_REF_PROPERTY);
-            Activity activity = new Activity(label, ref);
-            schedule.setActivities(Lists.newArrayList(activity));
+        }
+        if (node.has(TIMES_PROPERTY)) {
+            ArrayNode array = (ArrayNode)node.get(TIMES_PROPERTY);
```

Comment:
You'll want to use node.hasNonNull() instead of node.has(). Also, this code will throw a ClassCastException if times isn't an array. (This may or may not be handled by Jackson.) Also, you don't actually need to cast it to an ArrayNode, since JsonNode has a get() method. If this node is null, or if any of its children are null, this will throw a NullPointerException. Lastly, if any of the children are a malformatted LocalTime, this will throw some kind of RuntimeException.

All of this can be avoided by using Jackson directly instead of a custom serializer.